src/actions/getBuyNowListing.ts

```diff
@@ -0,0 +1,63 @@
+import { ethers } from "ethers";
+import {
+  getERC721Contract,
+  getZAuctionContract,
+  getZnsHubContract,
+} from "../contracts";
+import { BuyNowListing, Config } from "../types";
+import { Maybe, getLogger } from "../utilities";
+
+const logger = getLogger("actions:getBuyNowListing");
+
+export const getBuyNowListing = async (
+  tokenId: string,
+  config: Config
+): Promise<Maybe<BuyNowListing>> => {
+  if (!tokenId) throw Error("Must provide a valid tokenId");
```

Comment:
there are more guard clauses we can add in here to avoid additional method calls - we could pass in a non-null string for tokenId which would return the zero address on line 28, which should give us enough information to halt execution and invalidate input